deployment/docker/get_filter_messages/filter_msg_retriever.py

```diff
@@ -0,0 +1,91 @@
+# Python Imports
+import argparse
+import logging
+import urllib
+import requests
+import socket
+import time
+from concurrent.futures import ProcessPoolExecutor, as_completed
+from typing import Dict, List, Optional
+
+logging.basicConfig(level=logging.INFO)
+
+
+def resolve_dns(address: str) -> str:
+    start_time = time.time()
+    name, port = address.split(":")
+    ip_address = socket.gethostbyname(name)
+    elapsed = (time.time() - start_time) * 1000
+    logging.info(f"{address} DNS Response took {elapsed} ms")
+    logging.info(f"Talking with {address}, ip address: {ip_address}")
+
+    return f"{ip_address}:{port}"
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(description="Waku filter message retriever")
+    parser.add_argument('-c', '--contentTopic', type=str, help='Content topic', default="/my-app/1/dst/proto")
+    parser.add_argument('-n', '--numNodes', type=int, help='Number of filter nodes to get messages from', default=1)
+
+    return parser.parse_args()
+
+
+def fetch_all_messages(base_url: str, headers: Dict, address: str) -> Optional[List[str]]:
+    response = requests.get(base_url, headers=headers)
+    if response.status_code != 200:
+        logging.error(f"Error fetching data: {response.status_code}")
+        logging.error(response.text)
+        return None
```

Comment:
Double checking with you if it'd make more sense to use `Result` rather than `Optional`.

Reply:
It is a script on a standalone docker image, I would prefer to keep it as lightweight and simple as possible. 
Also, the functionality will remain rather small. We do not really benefit from `Result`. If it was in the core of the project, I would definetly use `Result`.